### PR TITLE
Fixing Issue#105

### LINKED ISSUE #105: pyEpiabm - read in parameters from .json file
Currently all stored in `pyEpiabm.core.parameters.py`

### PR DESCRIPTION
**Pull request by Saketkr21**
## Summary

Previously all parameters are stored in `pyEpiabm.core.parameters.py` file. As described in Issue #105 , It should be read from **json** file. I ran for loop on the file and added it as required class attributes.

## Further work

It converts list as numpy array and leave rest as it is. Not sure what might be edge case here.

## Checklist

- [x] All new functions have docstrings in the correct style
- [x] I've verified the complete docs build locally without errors
- [x] I've maintained 100% coverage (please mention any 'no cover' annotations explicitly)
- [x] I've unit-tested all new methods directly

- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [x] I have updated the wiki with new parameters/model functionality

## Closing issues
#Comment
`closes #105`

### CODE REVIEW

**Comment by KCGallagher** [comment]
Thanks so much for doing this! I'm afraid we have quite strict tests in the Continuous Integration, but I'll have a look through these to get them passing.

I think we'll also want to be able to specify the location of the parameters file, and then load that into the testing framework (as well as unit testing the .json importing method itself) but I'll have a look at that now

**Comment by KCGallagher** [comment]
> > Thanks so much for doing this! I'm afraid we have quite strict tests in the Continuous Integration, but I'll have a look through these to get them passing.
> > I think we'll also want to be able to specify the location of the parameters file, and then load that into the testing framework (as well as unit testing the .json importing method itself) but I'll have a look at that now
> 
> Thank you for your active response and encouragement. I am glad that I was able to contribute to my First Open Source Project. It will take time for me to understand code coverage and unit testing, But thank you for fixing CI/CD testing and finally achieving 100% coverage. I am really happy that I was able to give a small contribution to your project. Thanks !!

@Saketkr21 You're very welcome - I hope you enjoyed it, and let me know if you want to be involved further with this project! We'd really like to get external collaborators involved more widely, so can I ask how you found our project?

File: pyEpiabm/pyEpiabm/core/parameters.py [hunk 1]
```diff
@@ -16,247 +16,23 @@ class __Parameters:
...
-                                         0.47651]
-            }
-            self.host_progression_lists = defaultdict(int, host_prog_coeff)
+            parameters_file = open(config_file_path, "r")
+            parameters_str = parameters_file.read()
+            parameters = json.loads(parameters_str)
+            for key, value in parameters.items():
+                if isinstance(value, list):
+                    value = f'np.array({value})'
+                exec(f"self.{key} = {value}")
```
Comment (rcw5890) [inline pyEpiabm/pyEpiabm/core/parameters.py:31]: Is this really how to do json -> numpy

**Review by rcw5890 — CHANGES_REQUESTED** [review]

File: pyEpiabm/pyEpiabm/core/parameters.py (hunk 1 above)
Comment (rcw5890) [reply pyEpiabm/pyEpiabm/core/parameters.py:31]: Let's try `setattr`

**Review by rcw5890 — APPROVED** [review]

**Comment by Saketkr21** [comment]
> > > Thanks so much for doing this! I'm afraid we have quite strict tests in the Continuous Integration, but I'll have a look through these to get them passing.
> > > I think we'll also want to be able to specify the location of the parameters file, and then load that into the testing framework (as well as unit testing the .json importing method itself) but I'll have a look at that now
> > 
> > 
> > Thank you for your active response and encouragement. I am glad that I was able to contribute to my First Open Source Project. It will take time for me to understand code coverage and unit testing, But thank you for fixing CI/CD testing and finally achieving 100% coverage. I am really happy that I was able to give a small contribution to your project. Thanks !!
> 
> @Saketkr21 You're very welcome - I hope you enjoyed it, and let me know if you want to be involved further with this project! We'd really like to get external collaborators involved more widely, so can I ask how you found our project?

I found out about this project by searching globally all open issues in python language. Since I am a beginner, I am looking for more first_time_issue which are easy and fast to contribute to Open Source Community at this time.

**Pull request merged**